src/functionalTest/kotlin/com/autonomousapps/utils/AndroidProject.kt

```diff
@@ -29,6 +29,57 @@ fun libraryFactory(projectDir: File, librarySpec: LibrarySpec): Module {
     }
 }
 
+abstract class BaseGradleProject(val projectDir: File): Module {
+    override val dir = projectDir.also { it.mkdirs() }
+    fun withBuildFile(content: String, name: String = "build.gradle") {
+        projectDir.resolve(name).also { it.parentFile.mkdirs() }.writeText(content.trimIndent())
+    }
+    fun withFile(relativePath: String, content: String ) {
+        projectDir.resolve(relativePath).also { it.parentFile.mkdirs() }.writeText(content.trimIndent())
+    }
+}
+
+abstract class AndroidGradleProject(projectDir: File): BaseGradleProject(projectDir)  {
+    val mainDir = projectDir.resolve("src/main").also { it.mkdirs() }
+    val mainJavaSrcDir = mainDir.resolve("java").also { it.mkdirs() }
+    val resDir = mainDir.resolve("res").also { it.mkdirs() }
+
+    fun withManifestFile(content: String, relativePath: String = "AndroidManifest.xml"  ) {
```

Comment:
Shouldn't those be `= apply { }` to have a builder pattern?

Reply:
I don't understand your comment @pavlospt. Can you be more precise please, I am still new to kotlin.

Reply:
```suggestion
    fun withManifestFile(content: String, relativePath: String = "AndroidManifest.xml"  )  = apply {
```
This will return the same instance of the class so that you can call more functions on that builder. Think of it like you would do in Java:

```java
Builder withManifestFile(...) { 
   // something here
   return this;
}
```

Reply:
Thx @pavlospt, I agree it would be nicer but then we need to have a real builder.